pulpcore/cli/common/openapi.py

```diff
@@ -164,54 +170,117 @@ def extract_params(
+        if not isinstance(value, Dict):
+            raise OpenAPIError(_("'{name}' is expected to be an object.").format(name=name))
+        properties = schema.get("properties")
+        if properties is not None:
+            for property_name, property_value in value.items():
+                property_schema = properties.get(property_name)
+                if not property_schema:
+                    raise OpenAPIError(
+                        _("Unexpected property '{property_name}' for '{name}' provided.").format(
+                            name=name, property_name=property_name
+                        )
+                    )
+                value[property_name] = self.validate_schema(
+                    property_schema, property_name, property_value
+                )
         if "required" in schema:
-            missing_fields = set(schema["required"]) - set(body.keys()) - set(uploads.keys())
-            if missing_fields:
+            missing_properties = set(schema["required"]) - set(value.keys())
+            if missing_properties:
                 raise OpenAPIError(
-                    _("Required field(s) '{missing_fields}' missing.").format(
-                        missing_fields=missing_fields
+                    _("Required properties(s) '{missing_properties}' of '{name}' missing.").format(
+                        name=name, missing_properties=missing_properties
                     )
                 )
+        return value
+
+    def validate_array(self, schema: Any, name: str, value: Any) -> List[Any]:
+        if not isinstance(value, List):
+            print(value)
+            raise OpenAPIError(_("'{name}' is expected to be a list.").format(name=name))
+        item_schema = schema["items"]
+        return [self.validate_schema(item_schema, name, item) for item in value]
+
+    def validate_string(self, schema: Any, name: str, value: Any) -> Union[str, UploadType]:
+        schema_format = schema.get("format")
+        if schema_format == "date":
+            if not isinstance(value, datetime.date):
+                raise OpenAPIError(_("'{name}' is expected to be a date.").format(name=name))
+            return value.strftime(ISO_DATE_FORMAT)
+        elif schema_format == "date-time":
+            if not isinstance(value, datetime.datetime):
+                raise OpenAPIError(_("'{name}' is expected to be a datetime.").format(name=name))
+            return value.strftime(ISO_DATETIME_FORMAT)
+        elif schema_format == "binary":
+            if not isinstance(value, (bytes, BufferedReader)):
+                print(repr(value))
+                raise OpenAPIError(_("'{name}' is expected to be binary.").format(name=name))
+            return value
+        else:
+            if not isinstance(value, str):
+                raise OpenAPIError(_("'{name}' is expected to be a string.").format(name=name))
+            return value
```

Comment:
Should we be concern about the `byte` format? https://swagger.io/specification/#data-types. The comment says base64 encoded characters so maybe nothing else needs to be added, but I'm not sure.

Reply:
base64 could be checked for sure. On the other hand pulp does not use all the formats, so we could add a lot of untested code here.